types/target.go

```diff
@@ -42,6 +42,11 @@ func (x Target) Cmp(y Target) int {
 	return x.Int().Cmp(y.Int())
 }
 
+// Difficulty returns the difficulty associated with a given target.
+func (t Target) Difficulty() Currency {
+	return NewCurrency(new(big.Int).Div(RootDepth.Int(), t.Int()))
+}
```

Comment:
`Currency` should probably be renamed to `UInt` or something. `types.UInt` should be pretty informative. Maybe `SafeUInt`